bin/hdfcoinc/pycbc_fit_sngls_binned

```diff
@@ -27,27 +27,14 @@ from scipy.stats import kstest
 
 from pycbc import io, events, bin_utils, results
 from pycbc.events import trigger_fits as trstats
+from pycbc.events.stat import sngl_statistic_dict
 import pycbc.version
 
 #### DEFINITIONS AND FUNCTIONS ####
 
-stat_dict = {
-    "new_snr"       : events.newsnr,
-    "effective_snr" : events.effsnr,
-    "snr"           : lambda snr, rchisq : snr,
-    "snronchi"      : lambda snr, rchisq : snr / (rchisq ** 0.5),
-    "newsnr_sgveto"      : events.newsnr_sgveto,
-}
-
-def get_stat(statchoice, snr, rchisq, sgchisq, fac):
-    if fac is not None:
-        if statchoice not in ['new_snr', 'effective_snr']:
-            raise RuntimeError("Can't use --stat-factor with this statistic!")
-        return stat_dict[statchoice](snr, rchisq, fac)
-    elif statchoice == 'newsnr_sgveto':
-        return stat_dict[statchoice](snr, rchisq, sgchisq)
-    else:
-        return stat_dict[statchoice](snr, rchisq)
+def get_stat(statchoice, trigs):
+    stat_instance = sngl_statistic_dict[statchoice]([])
```

Comment:
add comment to this line e.g. 'initialize the stat class with an empty file list - in general could also feed it precalculated files'